contrib/terraform/openstack/modules/compute/main.tf

```diff
@@ -596,6 +596,7 @@ resource "openstack_compute_instance_v2" "k8s_nodes" {
   user_data         = each.value.cloudinit != null ? templatefile("${path.module}/templates/cloudinit.yaml.tmpl", {
     extra_partitions = each.value.cloudinit.extra_partitions
   }) : data.cloudinit_config.cloudinit.rendered
+  security_groups = var.port_security_enabled ? local.worker_sec_groups : null
```

Comment:
is this possibly due to a bug in kubespray?

Reply:
I was wondering the same thing, and after some reading I concluded that this likely is a problem that merits another PR or at least further discussion. In summary, I think it was an oversight when [removing port definitions](https://github.com/zonca/jetstream_kubespray/commit/87a36c00074e1702d199552f3067c156eca39b01) and then [fixing the broken security groups](https://github.com/zonca/jetstream_kubespray/commit/323ede0d96409a48a9e84df517e96b1fed377389) in a future commit. Since we hadn't used the `k8s_nodes` resource, it was never updated.

I think that series of commits was done to force Terraform to add instances to the auto_allocated_network. I would suggest we find a way to accomplish this without removing the ports resources so that we diverge as little as possible from "vanilla" Kubespray. This would make updating easier.

I can open a new issue that describes what I think the issue is more in depth and work on this when I have time?

Reply:
yes, sure, thanks, keep it low priority, maybe we can reconsider this when we update kubespray the next time